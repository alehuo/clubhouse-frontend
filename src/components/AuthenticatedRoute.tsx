import React from "react";
import { Redirect, Route, RouteProps } from "react-router-dom";

interface AuthenticatedRouteProps {
  component: React.ComponentType;
  isAuthenticated: boolean;
}
// TODO: Proper generic typings
const AuthenticatedRoute: React.SFC<AuthenticatedRouteProps & RouteProps> = ({
  component: Component,
  isAuthenticated,
  // FIXME: TSLint / TypeScript bug.
  // The object spread operator confuses the linter as a trailing comma is not allowed after the rest operator
  // @ts-ignore
  ...rest,
}) => (
  <Route
    {...rest}
    render={(props) => {
      return isAuthenticated ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: "/login",
            state: { from: props.location },
          }}
        />
      );
    }}
  />
);

export { AuthenticatedRoute };
