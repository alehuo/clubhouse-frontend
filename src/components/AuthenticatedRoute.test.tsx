import { shallow } from "enzyme";
import React from "react";
import { AuthenticatedRoute } from "./AuthenticatedRoute";

it("AuthenticatedRoute matches snapshot (not authenticated)", () => {
  const cmpnt = shallow(
    <AuthenticatedRoute isAuthenticated={false} component={() => <div>Hello</div>}/>,
  );
  expect(cmpnt).toMatchSnapshot();
});

it("AuthenticatedRoute matches snapshot (authenticated)", () => {
  const cmpnt = shallow(
    <AuthenticatedRoute isAuthenticated={true} component={() => <div>Hello</div>}/>,
  );
  expect(cmpnt).toMatchSnapshot();
});
