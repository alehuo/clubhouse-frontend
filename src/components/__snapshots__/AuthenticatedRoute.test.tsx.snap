// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticatedRoute matches snapshot (authenticated) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthenticatedRoute
    component={[Function]}
    isAuthenticated={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`AuthenticatedRoute matches snapshot (not authenticated) 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AuthenticatedRoute
    component={[Function]}
    isAuthenticated={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
